src/basics.md

```diff
@@ -805,6 +806,45 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-hmac]: #ex-hmac
+<a name="ex-hmac"></a>
+## Sign and verify a message with HMAC digest
+
+[![ring-badge]][ring] [![data-encoding-badge]][data-encoding] [![cat-cryptography-badge]][cat-cryptography]
+
+Creates a [`hmac::Signature`] of a string then verifies the signiture is correct.
+
+```rust
+extern crate ring;
+#
+# #[macro_use]
+# extern crate error_chain;
```

Comment:
for the sake of consistency with other examples i would suggest to put error_chain as the first `extern crate` without newlines below